src/Domain/Creators/SphericalShells.hpp

```diff
@@ -0,0 +1,260 @@
+
+/// \note This domain will use a spherical harmonic basis in the angular
+/// directions.  It cannot be used with subcell.
+///
+/// \see Sphere for a spherical domain compatible with subcell
+///
+/// This domain creator offers one grid anchor "Center" at the origin.
+///
+/// #### Time dependent maps
+/// There are two ways to add time dependent maps to the SphericalShells domain
+/// creator. In the input file, these are specified under the
+/// `TimeDependentMaps:` block.
+///
+/// ##### TimeDependence
+/// You can use a simple TimeDependence (e.g.
+/// `domain::creators::time_dependence::UniformTranslation` or
+/// `domain::creators::time_dependence::RotationAboutZAxis`) to add time
+/// dependent maps. This method will add the same maps to all blocks in the
+/// domain.
+///
+/// ##### Hard-coded time dependent maps
+/// The SphericalShells domain creator also has the option to use some hard
+/// coded time dependent maps that may be useful in certain scenarios. This
+/// method adds the maps in `domain::creators::sphere::TimeDependentMapOptions`
+/// to the domain. Currently, the first (inner-most) shell has maps between
+/// `Frame::Grid`, `Frame::Distorted`, and `Frame::Inertial` while all
+/// subsequent shells only have maps between `Frame::Grid` and
+/// `Frame::Inertial`.
+///
+/// ##### None
+/// To not have any time dependent maps, pass a `std::nullopt` as the
+/// appropriate argument in the constructor. In the input file, simply have
+/// `TimeDependentMaps: None`.
+class SphericalShells : public DomainCreator<3> {
+ public:
+  using maps_list =
+      tmpl::append<tmpl::list<domain::CoordinateMap<
+                       Frame::BlockLogical, Frame::Inertial,
+                       domain::CoordinateMaps::ProductOf2Maps<
+                           domain::CoordinateMaps::Interval,
+                           domain::CoordinateMaps::Identity<2>>,
+                       domain::CoordinateMaps::SphericalToCartesianPfaffian>>,
+                   typename sphere::TimeDependentMapOptions::maps_list>;
+
+  struct InnerRadius {
+    using type = double;
+    static constexpr Options::String help = {
+        "Inner radius of the spherical shells."};
+  };
+
+  struct OuterRadius {
+    using type = double;
+    static constexpr Options::String help = {
+        "Outer radius of the spherical shells."};
+  };
+
+  struct InitialRadialRefinement {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "Initial radial refinement level."};
+  };
+
+  struct InitialNumberOfRadialGridPoints {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "Initial number of radial grid points."};
+  };
+
+  struct InitialSphericalHarmonicL {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "Initial spherical harmonic resolution specified as the highest "
+        "spherical harmonic represented on the grid."};
+  };
```

Comment:
I think we should have a lower bound of 6 or 8 on this. I've never ran a stable simulation with an ell_max lower than that, even for spherically symmetric scalar wave data